price-feeder/config/currency_provider_tracker.go

```diff
@@ -0,0 +1,175 @@
+		pairs:               pairs,
+		coinIDSymbolMap:     map[string]string{},
+		CurrencyProviders:   map[string][]string{},
+		CurrencyProviderMin: map[string]int{},
+	}
+
+	if err := currencyProviderTracker.setCoinIDSymbolMap(); err != nil {
+		return nil, err
+	}
+
+	if err := currencyProviderTracker.setCurrencyProviders(); err != nil {
+		return nil, err
+	}
+
+	currencyProviderTracker.setCurrencyProviderMin()
+
+	go currencyProviderTracker.trackCurrencyProviders(ctx)
+
+	return currencyProviderTracker, nil
+}
+
+func (t *CurrencyProviderTracker) getCurrencyProviders() map[string][]string {
+	return t.CurrencyProviders
+}
+
+func (t *CurrencyProviderTracker) getCurrencyProviderMin(base string) int {
+	return t.CurrencyProviderMin[base]
+}
+
+func (t *CurrencyProviderTracker) logCurrencyProviders() {
+	for currency, providers := range t.CurrencyProviders {
+		t.logger.Info().Msg(fmt.Sprintf("Providers supporting %s: %v", currency, providers))
+	}
+}
+
+// setCoinIDSymbolMap gets list of assets on coingecko to cross reference coin symbol to id.
+func (t *CurrencyProviderTracker) setCoinIDSymbolMap() error {
+	resp, err := http.Get(coinGeckoRestHost + coinGeckoRestPath + coinGeckoListEndpoint)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	var listResponse []CoinGeckoCoinList
+	if err := json.NewDecoder(resp.Body).Decode(&listResponse); err != nil {
+		return err
+	}
+
+	for _, coin := range listResponse {
+		t.coinIDSymbolMap[coin.Symbol] = coin.ID
+	}
+
+	return nil
+}
+
+// setCurrencyProviders queries CoinGecko's tickers endpoint to get all the exchanges that
+// support each price feeder currency pair and store it in the CurrencyProviders map
+func (t *CurrencyProviderTracker) setCurrencyProviders() error {
+	for _, pair := range t.pairs {
+		pairBaseID := t.coinIDSymbolMap[strings.ToLower(pair.Base)]
+		resp, err := http.Get(coinGeckoRestHost + coinGeckoRestPath + pairBaseID + "/" + coinGeckoTickersEndpoint)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		var tickerResponse CoinGeckoCoinTickerResponse
+		if err = json.NewDecoder(resp.Body).Decode(&tickerResponse); err != nil {
+			return err
+		}
+
+		for _, ticker := range tickerResponse.Tickers {
+			if ticker.Target == pair.Quote {
+				t.CurrencyProviders[pair.Base] = append(t.CurrencyProviders[pair.Base], ticker.Market.Name)
+			}
+		}
+	}
+
+	return nil
+}
+
+// setCurrencyProviderMin will set the minimum amount of providers for each currency
+// to the amount of exchanges that support them if its less than 3. Otherwise it is
+// set to 3 providers.
+func (t *CurrencyProviderTracker) setCurrencyProviderMin() {
+	for base, exchanges := range t.getCurrencyProviders() {
+		if len(exchanges) < 3 {
+			t.CurrencyProviderMin[base] = len(exchanges)
+		} else {
+			t.CurrencyProviderMin[base] = 3
+		}
+	}
+}
+
+// trackCurrencyProviders resets CurrencyProviders map and logs out supported
+// exchanges for each currency every 24 hours
```

Comment:
As a general rule if it's a comment as a complete sentence, add a period - just a nitpick 
```suggestion
// trackCurrencyProviders resets CurrencyProviders map and logs supported
// exchanges for each currency every 24 hours.
```